packages/react-dom-bindings/src/server/ReactDOMServerFormatConfig.js

```diff
@@ -2001,65 +2034,172 @@ export function writeEndSegment(
 //   }
 // }
 //
-// function completeBoundary(suspenseBoundaryID, contentID) {
-//   // Find the fallback's first element.
-//   const suspenseIdNode = document.getElementById(suspenseBoundaryID);
-//   const contentNode = document.getElementById(contentID);
-//   // We'll detach the content node so that regardless of what happens next we don't leave in the tree.
-//   // This might also help by not causing recalcing each time we move a child from here to the target.
-//   contentNode.parentNode.removeChild(contentNode);
-//   if (!suspenseIdNode) {
-//     // The user must have already navigated away from this tree.
-//     // E.g. because the parent was hydrated. That's fine there's nothing to do
-//     // but we have to make sure that we already deleted the container node.
-//     return;
+// const insertStyles = (function() {
+//   const resourceMap = new Map();
+//   const precedences = new Map();
+//   // We omit rel stylesheet because only stylesheets should have data-prec attribute
+//   // and we can concievably use another kind of link to act as a placeholder for a
+//   // precedence that does not yet have any resources.
+//   let lastResource, node;
+//   let thisDocument = document;
+
+//   // Seed the precedence list with existing resources
+//   let nodes = thisDocument.querySelectorAll('link[data-rprec]');
```

Comment:
This takes a snapshot of this once, but what happens if the client hydrates a sibling before this and start inserting links? Fiber can discover newly inserted links but how does the Fizz runtime discover something that Fiber inserted.

Is there a case where it matters given that we don't guarantee sibling order?

Seems like you could end up inserting something in mixed precedence order though.

It might make sense to just query every time even though it's slower. It might help simplify the code too.